backend/hct_mis_api/aurora/admin.py

```diff
@@ -205,17 +205,20 @@ def fetch_aurora(self, request: HttpRequest) -> HttpResponse:
         if request.method == "POST":
             form = FetchForm(request.POST)
             aurora_token = request.user.custom_fields.get("aurora_token")
-            if "detect" in request.POST:
-                ctx["metadata"] = get_metadata(aurora_token)
-            elif form.is_valid():
-                filters = {
-                    "id": form.cleaned_data.get("from_id", "") or "",
-                    "after": form.cleaned_data.get("after_date", "") or "",
-                }
-                if form.cleaned_data["registration"]:
-                    filters["registration"] = form.cleaned_data["registration"].source_id
-                info = fetch_records(aurora_token, form.cleaned_data["overwrite"], **filters)
-                ctx["info"] = info
+            try:
+                if "detect" in request.POST:
+                    ctx["metadata"] = get_metadata(aurora_token)
+                elif form.is_valid():
+                    filters = {
+                        "id": form.cleaned_data.get("from_id", "") or "",
+                        "after": form.cleaned_data.get("after_date", "") or "",
+                    }
+                    if form.cleaned_data["registration"]:
```

Comment:
This part will raise KeyError everytime when `registration` is not in cleaned_data